int64 top
int64 bottom
int64 left
int64 right